### ISSUE TITLE
vagrant up error

### ISSUE THREAD

**Issue by levaconline** [hashicorp/vagrant#13520]
### Ruby version
ruby -v
ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [x86_64-linux]

### Gems list and versions
gem list

*** LOCAL GEMS ***

abbrev (default: 0.1.2)
base64 (default: 0.2.0)
benchmark (default: 0.3.0)
bigdecimal (default: 3.1.5)
bundler (default: 2.5.16)
cgi (default: 0.4.1)
csv (default: 3.2.8)
date (default: 3.3.4)
debug (1.9.1)
delegate (default: 0.3.1)
did_you_mean (default: 1.6.3)
digest (default: 3.1.1)
drb (default: 2.2.0)
english (default: 0.8.0)
erb (default: 4.0.4 ruby)
error_highlight (default: 0.6.0)
etc (default: 1.4.3)
fcntl (default: 1.1.0)
fiddle (default: 1.1.2)
fileutils (default: 1.7.2)
find (default: 0.2.0)
forwardable (default: 1.3.3)
getoptlong (default: 0.2.1)
io-console (default: 0.7.1)
io-nonblock (default: 0.3.0)
io-wait (default: 0.3.1)
ipaddr (default: 1.2.6)
irb (default: 1.13.1)
json (default: 2.7.1)
logger (default: 1.6.0)
matrix (0.4.2)
minitest (5.22.3)
mutex_m (default: 0.2.0)
net-ftp (0.3.5)
net-http (default: 0.4.1)
net-imap (0.4.12)
net-pop (0.1.2)
net-protocol (default: 0.2.2)
net-smtp (0.5.0)
nkf (default: 0.1.3)
observer (default: 0.1.2)
open-uri (default: 0.4.1)
open3 (default: 0.2.1)
openssl (default: 3.2.0)
optparse (default: 0.4.0)
ostruct (default: 0.6.0)
pathname (default: 0.3.0)
power_assert (2.0.3)
pp (default: 0.5.0)
prettyprint (default: 0.2.0)
prime (0.1.2)
prism (default: 0.19.0)
pstore (default: 0.1.3)
psych (default: 5.1.2)
racc (default: 1.7.3)
rake (13.1.0)
rbs (3.4.4)
rdoc (default: 6.6.3.1)
readline (default: 0.0.4)
reline (default: 0.5.7)
resolv (default: 0.3.0)
resolv-replace (default: 0.1.1)
rexml (3.3.2)
rinda (default: 0.2.0)
rss (0.3.1)
ruby2_keywords (default: 0.0.5)
securerandom (default: 0.3.1)
set (default: 1.1.0)
shellwords (default: 0.2.0)
singleton (default: 0.2.0)
stringio (default: 3.1.1)
strscan (default: 3.0.9)
syntax_suggest (default: 2.0.0)
syslog (default: 0.1.2)
tempfile (default: 0.2.1)
test-unit (3.6.2)
time (default: 0.3.0)
timeout (default: 0.4.1)
tmpdir (default: 0.2.0)
tsort (default: 0.2.0)
typeprof (0.21.9)
un (default: 0.3.0)
uri (default: 0.13.1)
weakref (default: 0.1.3)
yaml (default: 0.3.0)
zlib (default: 3.1.1)

### Debug output
vagrant up    
/usr/lib/ruby/3.3.0/rubygems/specification.rb:2245:in `raise_if_conflicts': Unable to activate vagrant_cloud-3.1.1, because rexml-3.3.2 conflicts with rexml (~> 3.2.5) (Gem::ConflictError)
	from /usr/lib/ruby/3.3.0/rubygems/specification.rb:1383:in `activate'
	from /usr/lib/ruby/3.3.0/rubygems/core_ext/kernel_gem.rb:62:in `block in gem'
	from /usr/lib/ruby/3.3.0/rubygems/core_ext/kernel_gem.rb:62:in `synchronize'
	from /usr/lib/ruby/3.3.0/rubygems/core_ext/kernel_gem.rb:62:in `gem'
	from /opt/vagrant/embedded/gems/gems/vagrant-2.4.1/bin/vagrant:95:in `block (2 levels) in <main>'
	from /opt/vagrant/embedded/gems/gems/vagrant-2.4.1/bin/vagrant:94:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-2.4.1/bin/vagrant:94:in `block in <main>'
	from /opt/vagrant/embedded/gems/gems/vagrant-2.4.1/bin/vagrant:105:in `<main>'

### Expected behavior
Exocet to work. I know, problem is sometime in versions. But what versions of Ruby & Ruby gems i need to fit to Vagrant?

### Actual behavior
Not work. No one answers my questions.

#### Vagrant version
vagrant -v
Vagrant 2.4.1

#### Host operating system
Operating System: Arch Linux                      
          Kernel: Linux 6.11.5-arch1-1
    Architecture: x86-64

#### Guest operating system
Ubuntu 22.04.4 LTS (GNU/Linux 5.15.0-91-generic x86_64)

#### Steps to reproduce

1. yay -Syu
2. cd project
3.vagrant up

#### Vagrantfile

```ruby
# Copy-paste your Vagrantfile here. Remove any sensitive information such as passwords, authentication tokens, or email addresses.
```

 

> -*- mode: ruby -*-
>  vi: set ft=ruby :
> 
> require 'json'
> require 'yaml'
> 
> VAGRANTFILE_API_VERSION ||= "2"
> scriptDir = $scriptDir ||= File.expand_path("tools/scripts/homestead", File.dirname(__FILE__))
> confDir = $confDir ||= File.expand_path("vendor/laravel/homestead", File.dirname(__FILE__))
> 
> homesteadYamlPath = scriptDir + "/Homestead.yaml"
> homesteadJsonPath = scriptDir + "/Homestead.json"
> afterScriptPath = scriptDir + "/after.sh"
> customizationScriptPath = scriptDir + "/user-customizations.sh"
> druidScriptPath = scriptDir + "/druid.sh"
> mobileAppsScriptPath = scriptDir + "/mobile-apps.sh"
> aliasesPath = scriptDir + "/aliases"
> kafkaScriptPath = scriptDir + "/kafka.sh"
> kafkaPhpExtScriptPath = scriptDir + "/kafka-php-ext.sh"
> 
> require File.expand_path(confDir + '/scripts/homestead.rb')
> 
> Vagrant.require_version '>= 2.4'
> 
> Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
>     config.vm.provider "parallels" do |prl|
>       prl.check_guest_tools = false
>     end
> 
>     if File.exist? aliasesPath then
>         config.vm.provision "file", source: aliasesPath, destination: "/tmp/bash_aliases"
>         config.vm.provision "shell" do |s|
>             s.inline = "awk '{ sub(\"\r$\", \"\"); print }' /tmp/bash_aliases > /home/vagrant/.bash_aliases"
>         end
>     end
> 
>     if File.exist? homesteadYamlPath then
>         settings = YAML::load(File.read(homesteadYamlPath))
>     elsif File.exist? homesteadJsonPath then
>         settings = JSON::parse(File.read(homesteadJsonPath))
>     else
>         abort "Homestead settings file not found in " + File.dirname(__FILE__)
>     end
> 
>     Homestead.configure(config, settings)
> 
>     if File.exist? afterScriptPath then
>         config.vm.provision "after", type: "shell", path: afterScriptPath
>     end
> 
>     if File.exist? customizationScriptPath then
>         config.vm.provision "custom", type: "shell", path: customizationScriptPath
>     end
> 
>     if Vagrant.has_plugin?('vagrant-hostsupdater')
>         config.hostsupdater.remove_on_suspend = false
>         config.hostsupdater.aliases = settings['sites'].map { |site| site['map'] }
>     elsif Vagrant.has_plugin?('vagrant-hostmanager')
>         config.hostmanager.enabled = true
>         config.hostmanager.manage_host = true
>         config.hostmanager.aliases = settings['sites'].map { |site| site['map'] }
>     elsif Vagrant.has_plugin?('vagrant-goodhosts')
>         config.goodhosts.aliases = settings['sites'].map { |site| site['map'] }
>     end
> 
>     if Vagrant.has_plugin?('vagrant-notify-forwarder')
>         config.notify_forwarder.enable = true
>     end
> 
>     Define test and DatingVIP hosts

    ...

**Comment by markus-as** [comment]
The same error has already been described [here](https://github.com/hashicorp/vagrant/issues/13502).
The linked issue is already closed after there was a workaround, but I don't see the problem as solved yet.

There is a [possible workaround](https://github.com/hashicorp/vagrant/issues/13502#issuecomment-2346722112) documented which refers to the downgrade of ruby-rexml.

If you want to use the workaround please be aware of security issues. Among others this https://www.ruby-lang.org/en/news/2024/10/28/redos-rexml-cve-2024-49761/

**Comment by levaconline** [comment]
[markus-as](https://github.com/markus-as) I seen they closed issue without problem solved. In meantime problem grown up (and i posted issue again), Now, problem is no more only ruby-rexm, but whole new ruby version and gems incompatible with vagrant. Tried downgrade and it did not worked. So, problems only grows up. :(
For me, vagrant up still only "dream".

**Comment by markus-as** [comment]
I just installed [vagrant-git](https://aur.archlinux.org/packages/vagrant-git) ( Verson2.4.4.dev+000330.216835c0.0.g216835c03-1 ) for testing and I don't get the error anymore. I don't have the time to go into it further at the moment and see it as a workaround for me.

**Comment by chrisroberts** [comment]
The [releases site](https://releases.hashicorp.com/vagrant/2.4.3/) includes an arch specific package that can be used: https://releases.hashicorp.com/vagrant/2.4.3/vagrant-2.4.3-1-x86_64.pkg.tar.zst

It provides a fully isolated install which will prevent issues encountered with shared ruby environments. I have some work started on the internal dependency resolver for handling shared environments better, but it will be some time before it is active. 

Cheers!

**Comment by levaconline** [comment]
AUR included vagrant 2.4.2 and after yay -Syu all worked fine for me. So, v2.4.2 solved issue related to ruby 3.3.0 compatibility.
